### ISSUE TITLE
lets_gan_clean: didnt converge with 10000 epochs

### ISSUE THREAD

**Issue by bicepjai** [ctmakro/hellotensor#1]
How long does it take to converge ?

**Comment by ctmakro** [comment]
@bicepjai this might be due to an implementation change of `BatchNorm` in the library `canton`, but unfortunately at the moment, I don't have the time to verify. Please consider taking a look at the commits of <https://github.com/ctmakro/canton>. Sorry for the inconvenience.

since last time the code was updated, more improvements on the training of GAN were invented by the academic world. you might wanna check <https://arxiv.org/pdf/1705.10461.pdf> for an example.

Wish you good luck!
